core/src/test/scala/unit/kafka/log/LogSegmentsTest.scala

```diff
@@ -18,15 +18,19 @@ package kafka.log
 
 import java.io.File
 
+import kafka.utils.TestUtils
 import org.apache.kafka.common.TopicPartition
-import org.apache.kafka.common.utils.Time
+import org.apache.kafka.common.utils.{Time, Utils}
 import org.junit.jupiter.api.Assertions._
-import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.{AfterEach, BeforeEach, Test}
+
+import scala.collection.mutable
 
 class LogSegmentsTest {
 
   val topicPartition = new TopicPartition("topic", 0)
   var logDir: File = _
+  val segmentsBuffer: mutable.ArrayBuffer[LogSegments] = mutable.ArrayBuffer[LogSegments]()
```

Comment:
This buffer can be removed, as I described below by calling `LogSegments#close()` inside each test.